src/CommunityToolkit.Maui.SourceGenerators/Generators/TextColorToGenerator.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using CommunityToolkit.Maui.SourceGenerators.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+
+namespace CommunityToolkit.Maui.SourceGenerators.Generators;
+
+[Generator]
+class TextColorToGenerator : ISourceGenerator
+{
+	public void Execute(GeneratorExecutionContext context)
+	{
+		var textStyleClassNameList = new List<string>();
+
+		var mauiControlsAssemblySymbol = context.Compilation.SourceModule.ReferencedAssemblySymbols.Single(q => q.Name == "Microsoft.Maui.Controls");
+
+		foreach (var namedTypeSymbol in mauiControlsAssemblySymbol.GlobalNamespace.GetNamedTypeSymbols())
+		{
+			if (namedTypeSymbol.ImplementsInterfaceOrBaseClass("ITextStyle"))
+			{
+				textStyleClassNameList.Add(namedTypeSymbol.Name);
+			}
+		}
+
+		foreach (var textStyleClassName in textStyleClassNameList)
+		{
+			var textColorToBuilder = @"
```

Comment:
we could make this a constant, and replace all `textStyleClassName` with `{0}`, and here we just call  `string.Format(textColorToBuilder, textStyleClassName)`.

Reply:
Honest question - does that improve performance since we are still injecting the textStyleClassName string?

Reply:
I believe that will not improve the performance, since we will create a new string every time. My suggestion was to reduce the method lines, moving this string to another place

Reply:
@brminnick Since the Initialize method doesn't support any code generation we can create a collection with all Maui types that implement ITextStyle and generate the method for those types during the initialization. Let me know what do you think about it